medallion/config.py

```diff
@@ -0,0 +1,82 @@
+import json
+import pathlib
+
+import attr
+import environ
+import jsonmerge
+
+from . import backends
+
+DEFAULT_CONFFILE = "/etc/medallion.conf"
+DEFAULT_CONFDIR = "/etc/medallion.d/"
+
+
+@environ.config(prefix="TAXII")
+class TAXIIConfig(object):
+    max_page_size = environ.var(None)
+
+
+@environ.config(prefix="MEDALLION")
+class MedallionConfig(object):
+    backend = environ.group(backends.BackendConfig)
+    taxii = environ.group(TAXIIConfig)
+
+    @classmethod
+    def __strip(cls, dict_):
+        for k, v in tuple(dict_.items()):
+            if isinstance(v, dict):
+                cls.__strip(v)
+            if v is None or v == {}:
+                del dict_[k]
+
+    def as_dict(self):
+        dictified = attr.asdict(self)
+        self.__strip(dictified)
+        return dictified
+
+
+def load_config(conf_file=DEFAULT_CONFFILE, conf_dir=DEFAULT_CONFDIR):
+    # Load configuration from the default config file or a value passed in
+    if conf_file is not None:
+        conf_file_p = pathlib.Path(conf_file)
+        try:
+            config_data = json.load(conf_file_p.open())
+        except FileNotFoundError as exc:
+            if conf_file is not DEFAULT_CONFFILE:
+                raise exc
+            else:
+                config_data = dict()
```

Comment:
This can be undefined if we skip this when `conf_file is None`. I'm reworking it to be a bit more DRY and fix this bug.